cli/azd/internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,197 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/user"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/internal"
+	appinsightsexporter "github.com/azure/azure-dev/cli/azd/internal/telemetry/appinsights-exporter"
+	"github.com/benbjohnson/clock"
+	"github.com/gofrs/flock"
+	"github.com/microsoft/ApplicationInsights-Go/appinsights"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
+)
+
+const telemetryItemExtension = ".trn"
+const devInstrumentationKey = "d3b9c006-3680-4300-9862-35fce9ac66c7"
+const prodInstrumentationKey = ""
```

Comment:
Will these be hard coded in the source? Are we concerned with this being exposed and used to stream bogus data into our system?

Reply:
They will be hard coded. And it's pretty much guaranteed for a client application that the instrumentation key needs to be exposed. Example from [dotnet CLI](https://github.com/dotnet/sdk/search?q=InstrumentationKey)